### PR TITLE
Update livecell finetuning and evaluation scripts

### PR DESCRIPTION
**Pull request by anwai98**
This addresses the following:
- Fixes a bug in the `livecell_evaluation.py` top-level evaluation script.
- Refactors the `livecell_finetuning.py` top-level finetuning script, to make it modular
    - NOTE: This only unifies the floating snippets for training, and does not change any components.
- Refactors the `train_sam` function to expose some optional arguments.

### CODE REVIEW

File: micro_sam/training/training.py [hunk 1]
```diff
@@ -93,11 +92,13 @@ def train_sam(
     freeze: Optional[List[str]] = None,
     device: Optional[Union[str, torch.device]] = None,
     lr: float = 1e-5,
-    scheduler: Optional[_LRScheduler] = None,
     n_sub_iteration: int = 8,
     save_root: Optional[Union[str, os.PathLike]] = None,
     mask_prob: float = 0.5,
     n_iterations: Optional[int] = None,
+    scheduler_kwargs: Optional[Dict[str, Any]] = None,
```
Comment (constantinpape) [inline micro_sam/training/training.py:99]: It's correct to remove the scheduler since it needs the optimizer, but it would be nice to enable another type of scheduler. We could introduce another argument `scheduler_class`, which defaults to `ScheduleLROnPlateau`, but makes the choice of scheduler fully flexible.

File: micro_sam/training/training.py [hunk 2]
```diff
@@ -93,11 +92,13 @@ def train_sam(
...
+    export_path: Optional[Union[str, os.PathLike]] = None,
```
Comment (constantinpape) [inline micro_sam/training/training.py:100]: I don't want to integrate export here, because we will have more export options once the modelzoo export is integrated.

File: micro_sam/training/training.py [hunk 3]
```diff
@@ -163,28 +168,73 @@ def train_sam(
...
+        trainer_fit_params = {"epochs": n_epochs}
+    else:
+        trainer_fit_params = {"iterations": n_iterations}
+
+    if save_every_kth_epoch is not None:
+        trainer_fit_params["save_every_kth_epoch"] = save_every_kth_epoch
+
+    trainer.fit(**trainer_fit_params)
+
+    if export_path is not None:
```
Comment (constantinpape) [inline micro_sam/training/training.py:234]: Let's keep the export separate, see above.

File: finetuning/livecell_finetuning.py [hunk 4]
```diff
@@ -52,79 +48,30 @@ def finetune_livecell(args):
...
+        device=device,
+        lr=1e-5,
+        n_iterations=args.iterations,
+        save_root=args.save_root,
+        scheduler_kwargs=scheduler_params,
+        export_path=args.export_path,
+        save_every_kth_epoch=args.save_every_kth_epoch,
     )
-    trainer.fit(args.iterations, save_every_kth_epoch=args.save_every_kth_epoch)
-    if args.export_path is not None:
```
Comment (constantinpape) [inline finetuning/livecell_finetuning.py:119]: Let's keep the export here.

File: micro_sam/training/training.py (hunk 1 above)
Comment (anwai98) [reply micro_sam/training/training.py:102]: I agree, makes sense to expose it as well. Updated it.

File: micro_sam/training/training.py [hunk 5]
```diff
@@ -163,28 +170,65 @@ def train_sam(
     else:
         optimizer = torch.optim.Adam(model.parameters(), lr=lr)
 
-    if scheduler is None:
-        scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(
-            optimizer, mode="min", factor=0.9, patience=3, verbose=True
-        )
+    scheduler_params = {"mode": "min", "factor": 0.9, "patience": 3, "verbose": True}
```
Comment (constantinpape) [inline micro_sam/training/training.py:173]: This will fail if we have a different scheduler class, because it will also get these params.
Let's simplify this:
```
if scheduler_kwargs is None:
  schedulder_kwargs = {...}  # the current defaults
```
Hence, if the user passes the kwargs every option needs to be set.

Comment (anwai98) [reply micro_sam/training/training.py]: Fixed. Thanks for spotting!

File: micro_sam/training/training.py [hunk 6]
```diff
@@ -163,28 +170,63 @@ def train_sam(
     else:
         optimizer = torch.optim.Adam(model.parameters(), lr=lr)
 
-    if scheduler is None:
-        scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(
-            optimizer, mode="min", factor=0.9, patience=3, verbose=True
-        )
+    if scheduler_kwargs is None:
+        scheduler_params = {"mode": "min", "factor": 0.9, "patience": 3, "verbose": True}
```
Comment (constantinpape) [inline micro_sam/training/training.py:174]: This needs to be `scheduler_kwargs` now.

File: micro_sam/training/training.py [hunk 7]
```diff
@@ -163,28 +170,63 @@ def train_sam(
...
+
+    scheduler = scheduler_class(optimizer=optimizer, **scheduler_params)
```
Comment (constantinpape) [inline micro_sam/training/training.py:176]: kwargs, see prev. comment.

**Review by constantinpape — APPROVED** [review]
